git_machete/client.py

```diff
@@ -935,11 +935,14 @@ def print_line_prefix(branch_: LocalBranchShortName, suffix: str) -> None:
             if branch in self.up_branch:
                 print_line_prefix(branch, f"{utils.get_vertical_bar()} \n")
                 if opt_list_commits:
-                    if edge_color[branch] in (EscapeCodes.RED, EscapeCodes.DIM):
-                        commits: List[GitLogEntry] = self.__git.get_commits_between(fp_sha(branch), branch.full_name()) if fp_sha(branch) else []
+                    if not fp_sha(branch):
+                        # Rare case, but can happen e.g. due to reflog expiry.
+                        commits: List[GitLogEntry] = []
+                    elif edge_color[branch] == EscapeCodes.DIM:
+                        commits = []
```

Comment:
These 2 cases are very distinct (completely different reasons, even though the resulting commit list is empty for both), I wanted to keep them separate e.g. so that the comment can be applied to just one of them in line 939 above